### ISSUE TITLE
[Twitcasting] Download failing on protected videos

### ISSUE THREAD

**Issue by Spicadox** [yt-dlp/yt-dlp#5888]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Provide a description that is worded well enough to be understood

There is an issue trying to download Twitcast videos and more specifically password protected videos. The issue may appear to be an extractor error when searching through the request form.  Protected videos can not be downloaded when providing a password nor will unlocking the videos manually and using the cookies from the browser work anymore.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--cookies-from-browser', 'chrome', '--video-password', 'PRIVATE', '-o', 'test.mkv', 'https://twitcasting.tv/yuzukichococh/movie/661534618']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.11.0 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 1.1.1q  5 Jul 2022)
[debug] exe versions: ffmpeg N-107100-g6a7c6ee93f-g4d45f5acbd (setts), ffprobe N-107100-g6a7c6ee93f-g4d45f5acbd
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-9.1
[Cookies] Extracting cookies from chrome
[debug] Extracting cookies from: "C:\Users\test\AppData\Local\Google\Chrome\User Data\Default\Network\Cookies"
[debug] Found local state file at "C:\Users\test\AppData\Local\Google\Chrome\User Data\Local State"
[Cookies] Extracted 2285 cookies from chrome
[debug] cookie version breakdown: {'v10': 2285, 'other': 0, 'unencrypted': 0}
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [TwitCasting] Extracting URL: https://twitcasting.tv/yuzukichococh/movie/661534618
[TwitCasting] 661534618: Downloading webpage
ERROR: [TwitCasting] 661534618: This video is protected by a password, use the --video-password option
  File "C:\Users\test\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\common.py", line 674, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\test\AppData\Local\Programs\Python\Python311\Lib\site-packages\yt_dlp\extractor\twitcasting.py", line 107, in _real_extract
    raise ExtractorError('This video is protected by a password, use the --video-password option', expected=True)
```

**Comment by bashonly** [comment]
This patch works with the password-protected test case in the extractor. Maybe you want to test it with your example URL as well:
```diff
diff --git a/yt_dlp/extractor/twitcasting.py b/yt_dlp/extractor/twitcasting.py
index 735cb0bb0..d015f5b18 100644
--- a/yt_dlp/extractor/twitcasting.py
+++ b/yt_dlp/extractor/twitcasting.py
@@ -88,15 +88,18 @@ def _parse_data_movie_playlist(self, dmp, video_id):
     def _real_extract(self, url):
         uploader_id, video_id = self._match_valid_url(url).groups()
 
+        webpage, urlh = self._download_webpage_handle(url, video_id)
         video_password = self.get_param('videopassword')
         request_data = None
         if video_password:
             request_data = urlencode_postdata({
                 'password': video_password,
+                **self._hidden_inputs(webpage),
             }, encoding='utf-8')
-        webpage, urlh = self._download_webpage_handle(
-            url, video_id, data=request_data,
-            headers={'Origin': 'https://twitcasting.tv'})
+            webpage, urlh = self._download_webpage_handle(
+                url, video_id, data=request_data,
+                headers={'Origin': 'https://twitcasting.tv'},
+                note='Trying video password')
         if urlh.geturl() != url and request_data:
             webpage = self._download_webpage(
                 urlh.geturl(), video_id, data=request_data,
```

**Comment by Spicadox** [comment]
Tested a couple of videos and looks like it's working. Thanks!
```
[debug] Command-line config: ['-vU', '-F', '--video-password', 'PRIVATE', 'https://twitcasting.tv/yuzukichococh/movie/661534618']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b644025b] (win_exe)
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.0 (CPython AMD64 64bit) - Windows-10-10.0.22621-SP0 (OpenSSL 1.1.1q  5 Jul 2022)
[debug] exe versions: ffmpeg N-107100-g6a7c6ee93f-g4d45f5acbd (setts), ffprobe N-107100-g6a7c6ee93f-g4d45f5acbd
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15, mutagen-1.45.1, sqlite3-2.6.0, websockets-9.1
[debug] Proxy map: {}
[debug] Loaded 1740 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[TwitCasting] Extracting URL: https://twitcasting.tv/yuzukichococh/movie/661534618
[TwitCasting] 661534618: Downloading webpage
[TwitCasting] 661534618: Trying video password
[TwitCasting] 661534618: Downloading live info
[TwitCasting] 661534618: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 661534618:
ID  EXT RESOLUTION │   FILESIZE  TBR PROTO │ VCODEC       VBR ACODEC     ABR
────────────────────────────────────────────────────────────────────────────
615 mp4 unknown    │ ~490.34MiB 615k m3u8  │ avc1.64001f 615k mp4a.40.29  0k
```
I see, that's good to know. Thanks for the extra info.
